packages/server/src/synchronizer/pipeline/rules/config/index.ts

```diff
@@ -0,0 +1,82 @@
+import {pathExistsSync} from 'fs-extra'
+import {resolve} from 'path'
+import File from 'vinyl'
+
+import {through} from '../../../streams'
+import {Rule} from '../../../types'
+
+const isNextConfigPath = (p: string) => /next\.config\.(js|ts)/.test(p)
+
+/**
+ * Returns a Rule that manages converting from blitz.config.js to next.config.js
+ */
+const create: Rule = ({config, input}) => {
+  // Preconditions
+  const hasNextConfig = pathExistsSync(resolve(config.src, 'next.config.js'))
+  const hasBlitzConfig = pathExistsSync(resolve(config.src, 'blitz.config.js'))
+
+  if (hasNextConfig && !process.env.NOW_BUILDER) {
+    // TODO: Pause the stream and ask the user if they wish to have their configuration file renamed
+    const err = new Error(
+      'Blitz does not support next.config.js. Please rename your next.config.js to blitz.config.js',
+    )
+    err.name = 'NextConfigSupportError'
+    throw err
+  }
+
+  if (!hasBlitzConfig) {
+    // Assume a bare blitz config
+    input.write(
+      new File({
+        cwd: config.src,
+        path: resolve(config.src, 'blitz.config.js'),
+        contents: Buffer.from('module.exports = {};'),
+      }),
+    )
+  }
+
+  if (!hasNextConfig) {
+    input.write(
+      new File({
+        cwd: config.src,
+        path: resolve(config.src, 'next.config.js'),
+        contents: Buffer.from(`
+const {withBlitz} = require('@blitzjs/server');
+const config = require('./blitz.config.js');
+module.exports = withBlitz(config);
+        `),
+      }),
+    )
+  }
+
+  // No need to filter yet
+  const stream = through({objectMode: true}, (file: File, _, next) => {
+    if (!isNextConfigPath(file.path)) return next(null, file)
+    // Zeit now adds configuration needed for Now, like serverless target,
+    // so we need to keep and use that
+    if (process.env.NOW_BUILDER) {
```

Comment:
We need to support `VERCEL_BUILDER` here too. Or instead, just check for `next.config.js` and use that regardless of the env variable. They won't be able to run blitz in dev mode with that file, so I think this would be fine.

Reply:
Hmm probably easier at this point to check for either vars